### PR TITLE
Rework on cadc-tap extra features.

### PR DESCRIPTION
**Pull request by yeunga**

### CODE REVIEW

File: cadctap/cadctap/core.py [hunk 1]
```diff
@@ -105,10 +103,13 @@
 ALLOWED_TB_DEF_TYPES = {'VOSITable': 'text/xml',
                         'VOTable': 'application/x-votable+xml'}
 AUTH_OPTION_EXPLANATION = \
-    'If no authentication option is specified, cadc-tap will look in the\n'\
-    '~/.netrc file for the cadc-ccda.hia-iha.nrc-cnrc.gc.ca or canfar.net\n'\
-    'domain, and if found, will use the -n option. If not, cadc-tap will\n'\
-    'look for ~/.ssl/cadcproxy.pem file, and if found, will use the --cert\n'\
+    '\nTo obtain the host associated with a service, execute a subcommand\n'\
```
Comment (andamian) [inline cadctap/cadctap/core.py]: This blob repeats a few times. Please assign it to a variable and display that everywhere is necessary. That would be great for maintenance.

File: cadctap/setup.cfg [hunk 2]
```diff
@@ -36,7 +36,7 @@ edit_on_github = False
 github_project = opencadc/caom2tools 
 # version should be PEP386 compatible (http://www.python.org/dev/peps/pep-0386)
 install_requires = cadcutils>=1.1.15 clint>=0.5.1 python-magic>=0.4.15 requests>=2.18 requests-toolbelt>=0.8.0
-version = 0.8
+version = 0.9
```
Comment (andamian) [inline cadctap/setup.cfg]: Make the version 0.8.1 since it doesn't introduce major API changes and it's mostly bug fixes.

File: cadctap/setup.py [hunk 3]
```diff
@@ -55,6 +56,9 @@ def readme():
 for entry_point in entry_point_list:
     entry_points['console_scripts'].append('{0} = {1}'.format(entry_point[0],
                                                               entry_point[1]))
+# copy example_query.sql file to /tmp so that the examples in cadc-tap query -h work
```
Comment (andamian) [inline cadctap/setup.py]: I don't think it's worth doing this when most of the time is not going to be used.

File: cadctap/cadctap/core.py [hunk 4]
```diff
@@ -431,13 +432,17 @@ def _customize_parser(parser):
              DEFAULT_SERVICE_ID))
 
 
-def _get_subject_from_netrc():
+def _get_subject_from_netrc(service):
```
[the remark this answers is not in the document]
Comment (yeunga) [reply cadctap/cadctap/core.py]: Done

File: cadctap/cadctap/core.py [hunk 5]
```diff
@@ -549,15 +554,15 @@ def main_app(command='cadc-tap query'):
         '- Anonymously run a query string:\n'
         '      {0} -a -s tap "SELECT TOP 10 type FROM caom2.Observation"\n'
         '- Use certificate to run a query from a file:\n'
-        '      {0} -s tap -i ./cadctap/tests/data/example_query.sql'
+        '      {0} -s tap -i /tmp/cadctap/data/example_query.sql'
```
[the remark this answers is not in the document]
Comment (yeunga) [reply cadctap/cadctap/core.py]: Could you please elaborate? We still use the '-i' option, I suppose?

File: cadctap/cadctap/core.py (hunk 1 above)
Comment (yeunga) [reply cadctap/cadctap/core.py]: The phrase 'host associated with a service' occurs twice. Do we really want to define a variable for that?

File: cadctap/setup.cfg (hunk 2 above)
Comment (yeunga) [reply cadctap/setup.cfg]: Done

File: cadctap/setup.py (hunk 3 above)
Comment (yeunga) [reply cadctap/setup.py]: Removed.

File: cadctap/cadctap/tests/test_core.py [hunk 6]
```diff
@@ -508,6 +509,60 @@ def test_help(self):
...
+                                         MyExitError, MyExitError]))
+    def test_exit_on_exception(self):
+        """ Test the exit_on_exception function """
+
+        # test handling of 'Bad Request' server errors
+        with patch('sys.stderr', new_callable=StringIO) as stderr_mock:
+            expected_message = "source error message"
+            orig_ex = RuntimeError("Bad Request")
+            ex = exceptions.HttpException(expected_message, orig_ex)
+            try:
```
Comment (andamian) [inline cadctap/cadctap/tests/test_core.py]: No need to use try, except - just pass the exception to exit_on_exception

Comment (yeunga) [reply cadctap/cadctap/tests/test_core.py]: That's what I did initially, but python 3.4 did not like it.

**Pull request merged**